base/dict.jl

```diff
@@ -189,6 +189,12 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict)
+
+Create an empty dictionary with the same key and value types as `d`.
+"""
 similar{K,V}(d::Dict{K,V}) = Dict{K,V}()
 similar{K,V}(d::Dict, ::Type{Pair{K,V}}) = Dict{K,V}()
```

Comment:
Woops, I just realized you could also document the second method while you're at it. Not sure how to best do that; maybe adding a `T=typeof(keytype(d)=>valtype(d))` argument, but that's not terribly pretty.